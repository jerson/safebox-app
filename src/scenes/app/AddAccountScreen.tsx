import React, { useState, useRef } from 'react';
import {
  StyleSheet,
  View,
  ViewStyle,
  ScrollView,
  StatusBar
} from 'react-native';
import { SafeAreaView } from 'react-navigation';
import Colors from '../../modules/constants/Colors';
import TextInput, { TextInputRef } from '../../components/ui/TextInput';
import Button from '../../components/ui/Button';
import Container from '../../components/ui/Container';
import Content from '../../components/ui/Content';
import Size from '../../modules/dimensions/Size';
import useTextInput from '../../components/hooks/useTextInput';
import Client from '../../services/Client';
import {
  AddAccountRequest,
  Account,
  AccountSingle
} from '../../proto/services_pb';
import OpenPGP from 'react-native-fast-openpgp';

import Session from '../../services/Session';
import Strings from '../../modules/format/Strings';
import AlertMessage from '../../components/ui/AlertMessage';
import useAnimatedState from '../../components/hooks/useAnimatedState';
import SplitText from '../../components/ui/SplitText';
import useIconLabel from '../../components/hooks/useIconLabel';

import { useNavigation } from 'react-navigation-hooks';

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.grey1
  } as ViewStyle,
  scrollView: {
    flex: 1
  } as ViewStyle,
  safeArea: {
    flex: 1
  } as ViewStyle,
  textInput: {} as ViewStyle,
  textInputContainer: {
    marginBottom: 0
  } as ViewStyle,
  form: {
    width: 280,
    marginBottom: 60,
    alignSelf: 'center'
  } as ViewStyle,
  button: {
    marginTop: 20
  } as ViewStyle,
  splitView: {
    marginTop: 20
  } as ViewStyle
});

function AddAccountScreen() {
  const { replace } = useNavigation();

  const [isLoading, setIsLoading] = useState(false);
  const [error, setError] = useAnimatedState('');

  const labelRef = useRef<TextInputRef>(null);
  const hintRef = useRef<TextInputRef>(null);
  const usernameRef = useRef<TextInputRef>(null);
  const passwordRef = useRef<TextInputRef>(null);
  const repeatPasswordRef = useRef<TextInputRef>(null);

  const [label, labelProps] = useTextInput('');
  const [hint, hintProps] = useTextInput('');
  const [username, usernameProps] = useTextInput('');
  const [password, passwordProps] = useTextInput('');
  const [repeatPassword, repeatPasswordProps] = useTextInput('');

  const isValid = () => {
    const isValid = [!!label, !!username, !!password, !!repeatPassword].every(
      value => value
    );
    if (!isValid) {
      setError('Complete missing fields');
      return false;
    }
    if (password !== repeatPassword) {
      setError('Passwords must match');
      return false;
    }
    return isValid;
  };

  const encode = (input: string) => {
    return OpenPGP.encrypt(input, Session.getPublicKey());
  };

  const submit = async () => {
    try {
      const passwordEncoded = await encode(password);

      const account = new Account();
      account.setHint(hint);
      account.setUsername(username);
      account.setPassword(passwordEncoded);
      account.setLabel(label);

      const request = new AddAccountRequest();
      request.setAccount(account);

      const response = await Client.addAccount(request);

      const accountSingle = new AccountSingle();
      accountSingle.setId(response.getId());
      accountSingle.setHint(hint);
      accountSingle.setUsername(username);
      accountSingle.setLabel(label);

      replace('Account', { account: accountSingle });
      return;
    } catch (e) {
      const message = Strings.getError(e);
      setError(message);
    }
    setIsLoading(false);
  };
  const tryToSubmit = () => {
    if (!isValid()) {
      return;
    }

    setIsLoading(true);
    requestAnimationFrame(async () => {
      await submit();
    });
  };

  const labelExtraProps = useIconLabel(label);
  return (
    <Container style={styles.container}>
      <StatusBar
        animated
        barStyle={'dark-content'}
        backgroundColor={Colors.grey2}
      />
      <ScrollView
        keyboardShouldPersistTaps={'handled'}
        contentContainerStyle={{
          minHeight: Size.getVisibleTabScreenHeight()
        }}
        style={styles.scrollView}
      >
        <SafeAreaView style={styles.safeArea}>
          <Content center>
            <View style={styles.form}>
              {!!error && <AlertMessage message={error} />}
              <TextInput
                placeholder={'Label'}
                keyboardType={'default'}
                autoCapitalize={'sentences'}
                autoCorrect
                returnKeyType={'next'}
                containerStyle={styles.textInputContainer}
                style={styles.textInput}
                ref={labelRef}
                onSubmitEditing={() => {
                  hintRef.current && hintRef.current.focus();
                }}
                {...labelProps}
                {...labelExtraProps}
              />
              <TextInput
                icon={'feather'}
                placeholder={'Hint (optional)'}
                keyboardType={'default'}
                autoCapitalize={'sentences'}
                autoCorrect
                returnKeyType={'next'}
                containerStyle={styles.textInputContainer}
                style={styles.textInput}
                ref={hintRef}
                onSubmitEditing={() => {
                  usernameRef.current && usernameRef.current.focus();
                }}
                {...hintProps}
              />
              <SplitText
                type={'Default'}
                style={styles.splitView}
                title={'Sensitive credentials'}
              />
              <TextInput
                icon={'user'}
                placeholder={'Username'}
                keyboardType={'default'}
                autoCapitalize={'none'}
                autoCorrect={false}
                autoCompleteType={'off'}
                returnKeyType={'next'}
                containerStyle={styles.textInputContainer}
                style={styles.textInput}
                ref={usernameRef}
                onSubmitEditing={() => {
                  passwordRef.current && passwordRef.current.focus();
                }}
                {...usernameProps}
              />
              <TextInput
                icon={'lock'}
                placeholder={'Password'}
                secureTextEntry
                keyboardType={'default'}
                autoCapitalize={'none'}
                autoCorrect={false}
                autoCompleteType={'off'}
                returnKeyType={'next'}
                containerStyle={styles.textInputContainer}
                style={styles.textInput}
                ref={passwordRef}
                onSubmitEditing={() => {
                  repeatPasswordRef.current &&
                    repeatPasswordRef.current.focus();
                }}
                {...passwordProps}
              />
              <TextInput
                icon={'lock'}
                placeholder={'Repeat password'}
                secureTextEntry
                keyboardType={'default'}
                autoCapitalize={'none'}
                autoCorrect={false}
                autoCompleteType={'off'}
                returnKeyType={'done'}
                containerStyle={styles.textInputContainer}
                style={styles.textInput}
                ref={repeatPasswordRef}
                blurOnSubmit
                {...repeatPasswordProps}
              />

              <Button
                isLoading={isLoading}
                style={styles.button}
                typeColor={'primaryLight'}
                title={'Add secret account'}
                onPress={tryToSubmit}
              />
            </View>
          </Content>
        </SafeAreaView>
      </ScrollView>
    </Container>
  );
}

AddAccountScreen.navigationOptions = () => ({
  title: 'Add secret account'
});

export default AddAccountScreen;
