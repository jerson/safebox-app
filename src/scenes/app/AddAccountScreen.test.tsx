import React from 'react';
import * as renderer from 'react-test-renderer';
import AddAccountScreen from './AddAccountScreen';

const props: AddAccountScreen = {};
test('renders without crashing', () => {
  const snapshot = renderer.create(<AddAccountScreen {...props} />).toJSON();
  expect(snapshot).toBeTruthy();
});

test('render just views', () => {
  const snapshot = renderer.create(<AddAccountScreen {...props} />).toJSON();
  expect(snapshot).toMatchSnapshot();
});
