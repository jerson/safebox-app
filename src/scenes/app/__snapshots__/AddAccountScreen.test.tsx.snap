// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render just views 1`] = `
<_class
  style={
    Array [
      Object {
        "backgroundColor": "#fafdff",
        "flex": 1,
      },
      Object {
        "backgroundColor": "#fafdff",
      },
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "minHeight": 1220,
      }
    }
    keyboardShouldPersistTaps="handled"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <_class>
      <_class
        onLayout={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "flex": 1,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 20,
          }
        }
      >
        <_class
          style={
            Array [
              Object {
                "paddingHorizontal": 20,
              },
              Object {
                "justifyContent": "center",
              },
              Object {
                "flex": 1,
                "paddingTop": 20,
              },
              undefined,
            ]
          }
        >
          <_class
            onLayout={[Function]}
            style={
              Object {
                "paddingBottom": 0,
              }
            }
          >
            <_class
              style={
                Object {
                  "alignSelf": "center",
                  "marginBottom": 60,
                  "width": 280,
                }
              }
            >
              <_class
                style={
                  Array [
                    Object {
                      "flexDirection": "column",
                      "marginBottom": 20,
                      "marginTop": 10,
                      "position": "relative",
                    },
                    Object {
                      "marginBottom": 0,
                    },
                  ]
                }
              >
                <_class
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "paddingBottom": 4,
                      },
                      undefined,
                    ]
                  }
                />
                <_class
                  style={
                    Array [
                      Object {
                        "position": "relative",
                      },
                      undefined,
                    ]
                  }
                >
                  <TextInput
                    allowFontScaling={true}
                    autoCapitalize="sentences"
                    autoCorrect={true}
                    editable={true}
                    keyboardType="default"
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    onSubmitEditing={[Function]}
                    placeholder="Label"
                    placeholderTextColor="#65748b"
                    rejectResponderTermination={true}
                    returnKeyType="next"
                    selectionColor="#65748b"
                    style={
                      Array [
                        Object {
                          "backgroundColor": "#FFFFFF",
                          "borderColor": "#f2f4f9",
                          "borderRadius": 10,
                          "borderWidth": 1,
                          "color": "#38414e",
                          "fontFamily": "Nunito",
                          "fontSize": 15,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "paddingBottom": 10,
                          "paddingLeft": 10,
                          "paddingRight": 8,
                          "paddingTop": 10,
                          "textDecorationLine": "none",
                        },
                        Object {
                          "paddingLeft": 40,
                        },
                        false,
                        false,
                        Object {
                          "paddingRight": 40,
                        },
                        Object {},
                      ]
                    }
                    underlineColorAndroid="transparent"
                    value=""
                  />
                  <_class
                    style={
                      Object {
                        "bottom": 1,
                        "justifyContent": "center",
                        "left": 12,
                        "position": "absolute",
                        "top": 0,
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": undefined,
                            "fontSize": 12,
                          },
                          Array [
                            Object {
                              "color": "#65748b",
                              "fontSize": 20,
                            },
                            false,
                            false,
                            Object {
                              "color": "#65748b",
                            },
                          ],
                          Object {
                            "fontFamily": "Feather",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      ï‡Ÿ
                    </Text>
                  </_class>
                  <_class
                    style={
                      Array [
                        Object {
                          "position": "absolute",
                          "right": -60,
                        },
                        undefined,
                      ]
                    }
                  />
                </_class>
              </_class>
              <_class
                style={
                  Array [
                    Object {
                      "flexDirection": "column",
                      "marginBottom": 20,
                      "marginTop": 10,
                      "position": "relative",
                    },
                    Object {
                      "marginBottom": 0,
                    },
                  ]
                }
              >
                <_class
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "paddingBottom": 4,
                      },
                      undefined,
                    ]
                  }
                />
                <_class
                  style={
                    Array [
                      Object {
                        "position": "relative",
                      },
                      undefined,
                    ]
                  }
                >
                  <TextInput
                    allowFontScaling={true}
                    autoCapitalize="sentences"
                    autoCorrect={true}
                    editable={true}
                    keyboardType="default"
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    onSubmitEditing={[Function]}
                    placeholder="Hint (optional)"
                    placeholderTextColor="#65748b"
                    rejectResponderTermination={true}
                    returnKeyType="next"
                    selectionColor="#65748b"
                    style={
                      Array [
                        Object {
                          "backgroundColor": "#FFFFFF",
                          "borderColor": "#f2f4f9",
                          "borderRadius": 10,
                          "borderWidth": 1,
                          "color": "#38414e",
                          "fontFamily": "Nunito",
                          "fontSize": 15,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "paddingBottom": 10,
                          "paddingLeft": 10,
                          "paddingRight": 8,
                          "paddingTop": 10,
                          "textDecorationLine": "none",
                        },
                        Object {
                          "paddingLeft": 40,
                        },
                        false,
                        false,
                        Object {
                          "paddingRight": 40,
                        },
                        Object {},
                      ]
                    }
                    underlineColorAndroid="transparent"
                    value=""
                  />
                  <_class
                    style={
                      Object {
                        "bottom": 1,
                        "justifyContent": "center",
                        "left": 12,
                        "position": "absolute",
                        "top": 0,
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": undefined,
                            "fontSize": 12,
                          },
                          Array [
                            Object {
                              "color": "#65748b",
                              "fontSize": 20,
                            },
                            false,
                            false,
                            undefined,
                          ],
                          Object {
                            "fontFamily": "Feather",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      ï…¡
                    </Text>
                  </_class>
                  <_class
                    style={
                      Array [
                        Object {
                          "position": "absolute",
                          "right": -60,
                        },
                        undefined,
                      ]
                    }
                  />
                </_class>
              </_class>
              <_class
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                    },
                    Object {
                      "marginTop": 20,
                    },
                  ]
                }
              >
                <_class
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#9066f5",
                        "height": 1,
                        "left": 0,
                        "opacity": 0.5,
                        "position": "absolute",
                        "right": 0,
                        "top": 12,
                      },
                      Object {
                        "backgroundColor": "#c2d0e7",
                      },
                      false,
                    ]
                  }
                />
                <_class
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#4531cd",
                        "borderRadius": 10,
                        "paddingBottom": 4,
                        "paddingLeft": 8,
                        "paddingRight": 8,
                        "paddingTop": 6,
                      },
                      Object {
                        "backgroundColor": "#6781a6",
                      },
                      false,
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontFamily": "Nunito",
                          "fontSize": undefined,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                        },
                        Object {},
                        Object {
                          "color": "#FFFFFF",
                          "fontSize": 13,
                          "lineHeight": 14,
                          "minWidth": 20,
                          "textAlign": "center",
                        },
                      ]
                    }
                  >
                    Sensitive credentials
                  </Text>
                </_class>
              </_class>
              <_class
                style={
                  Array [
                    Object {
                      "flexDirection": "column",
                      "marginBottom": 20,
                      "marginTop": 10,
                      "position": "relative",
                    },
                    Object {
                      "marginBottom": 0,
                    },
                  ]
                }
              >
                <_class
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "paddingBottom": 4,
                      },
                      undefined,
                    ]
                  }
                />
                <_class
                  style={
                    Array [
                      Object {
                        "position": "relative",
                      },
                      undefined,
                    ]
                  }
                >
                  <TextInput
                    allowFontScaling={true}
                    autoCapitalize="none"
                    autoCompleteType="off"
                    autoCorrect={false}
                    editable={true}
                    keyboardType="default"
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    onSubmitEditing={[Function]}
                    placeholder="Username"
                    placeholderTextColor="#65748b"
                    rejectResponderTermination={true}
                    returnKeyType="next"
                    selectionColor="#65748b"
                    style={
                      Array [
                        Object {
                          "backgroundColor": "#FFFFFF",
                          "borderColor": "#f2f4f9",
                          "borderRadius": 10,
                          "borderWidth": 1,
                          "color": "#38414e",
                          "fontFamily": "Nunito",
                          "fontSize": 15,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "paddingBottom": 10,
                          "paddingLeft": 10,
                          "paddingRight": 8,
                          "paddingTop": 10,
                          "textDecorationLine": "none",
                        },
                        Object {
                          "paddingLeft": 40,
                        },
                        false,
                        false,
                        Object {
                          "paddingRight": 40,
                        },
                        Object {},
                      ]
                    }
                    underlineColorAndroid="transparent"
                    value=""
                  />
                  <_class
                    style={
                      Object {
                        "bottom": 1,
                        "justifyContent": "center",
                        "left": 12,
                        "position": "absolute",
                        "top": 0,
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": undefined,
                            "fontSize": 12,
                          },
                          Array [
                            Object {
                              "color": "#65748b",
                              "fontSize": 20,
                            },
                            false,
                            false,
                            undefined,
                          ],
                          Object {
                            "fontFamily": "Feather",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      ï‡¶
                    </Text>
                  </_class>
                  <_class
                    style={
                      Array [
                        Object {
                          "position": "absolute",
                          "right": -60,
                        },
                        undefined,
                      ]
                    }
                  />
                </_class>
              </_class>
              <_class
                style={
                  Array [
                    Object {
                      "flexDirection": "column",
                      "marginBottom": 20,
                      "marginTop": 10,
                      "position": "relative",
                    },
                    Object {
                      "marginBottom": 0,
                    },
                  ]
                }
              >
                <_class
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "paddingBottom": 4,
                      },
                      undefined,
                    ]
                  }
                />
                <_class
                  style={
                    Array [
                      Object {
                        "position": "relative",
                      },
                      undefined,
                    ]
                  }
                >
                  <TextInput
                    allowFontScaling={true}
                    autoCapitalize="none"
                    autoCompleteType="off"
                    autoCorrect={false}
                    editable={true}
                    keyboardType="default"
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    onSubmitEditing={[Function]}
                    placeholder="Password"
                    placeholderTextColor="#65748b"
                    rejectResponderTermination={true}
                    returnKeyType="next"
                    secureTextEntry={true}
                    selectionColor="#65748b"
                    style={
                      Array [
                        Object {
                          "backgroundColor": "#FFFFFF",
                          "borderColor": "#f2f4f9",
                          "borderRadius": 10,
                          "borderWidth": 1,
                          "color": "#38414e",
                          "fontFamily": "Nunito",
                          "fontSize": 15,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "paddingBottom": 10,
                          "paddingLeft": 10,
                          "paddingRight": 8,
                          "paddingTop": 10,
                          "textDecorationLine": "none",
                        },
                        Object {
                          "paddingLeft": 40,
                        },
                        false,
                        false,
                        Object {
                          "paddingRight": 40,
                        },
                        Object {},
                      ]
                    }
                    underlineColorAndroid="transparent"
                    value=""
                  />
                  <_class
                    style={
                      Object {
                        "bottom": 1,
                        "justifyContent": "center",
                        "left": 12,
                        "position": "absolute",
                        "top": 0,
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": undefined,
                            "fontSize": 12,
                          },
                          Array [
                            Object {
                              "color": "#65748b",
                              "fontSize": 20,
                            },
                            false,
                            false,
                            undefined,
                          ],
                          Object {
                            "fontFamily": "Feather",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      ï†‰
                    </Text>
                  </_class>
                  <_class
                    style={
                      Array [
                        Object {
                          "position": "absolute",
                          "right": -60,
                        },
                        undefined,
                      ]
                    }
                  />
                </_class>
              </_class>
              <_class
                style={
                  Array [
                    Object {
                      "flexDirection": "column",
                      "marginBottom": 20,
                      "marginTop": 10,
                      "position": "relative",
                    },
                    Object {
                      "marginBottom": 0,
                    },
                  ]
                }
              >
                <_class
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "paddingBottom": 4,
                      },
                      undefined,
                    ]
                  }
                />
                <_class
                  style={
                    Array [
                      Object {
                        "position": "relative",
                      },
                      undefined,
                    ]
                  }
                >
                  <TextInput
                    allowFontScaling={true}
                    autoCapitalize="none"
                    autoCompleteType="off"
                    autoCorrect={false}
                    blurOnSubmit={true}
                    editable={true}
                    keyboardType="default"
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    onSubmitEditing={[Function]}
                    placeholder="Repeat password"
                    placeholderTextColor="#65748b"
                    rejectResponderTermination={true}
                    returnKeyType="done"
                    secureTextEntry={true}
                    selectionColor="#65748b"
                    style={
                      Array [
                        Object {
                          "backgroundColor": "#FFFFFF",
                          "borderColor": "#f2f4f9",
                          "borderRadius": 10,
                          "borderWidth": 1,
                          "color": "#38414e",
                          "fontFamily": "Nunito",
                          "fontSize": 15,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "paddingBottom": 10,
                          "paddingLeft": 10,
                          "paddingRight": 8,
                          "paddingTop": 10,
                          "textDecorationLine": "none",
                        },
                        Object {
                          "paddingLeft": 40,
                        },
                        false,
                        false,
                        Object {
                          "paddingRight": 40,
                        },
                        Object {},
                      ]
                    }
                    underlineColorAndroid="transparent"
                    value=""
                  />
                  <_class
                    style={
                      Object {
                        "bottom": 1,
                        "justifyContent": "center",
                        "left": 12,
                        "position": "absolute",
                        "top": 0,
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": undefined,
                            "fontSize": 12,
                          },
                          Array [
                            Object {
                              "color": "#65748b",
                              "fontSize": 20,
                            },
                            false,
                            false,
                            undefined,
                          ],
                          Object {
                            "fontFamily": "Feather",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      ï†‰
                    </Text>
                  </_class>
                  <_class
                    style={
                      Array [
                        Object {
                          "position": "absolute",
                          "right": -60,
                        },
                        undefined,
                      ]
                    }
                  />
                </_class>
              </_class>
              <_class
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "#9066f5",
                    "borderColor": "#9066f5",
                    "borderRadius": 10,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "margin": 0,
                    "marginTop": 20,
                    "opacity": 1,
                    "padding": 12,
                    "zIndex": 1,
                  }
                }
                testID="button_touchable"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Nunito",
                        "fontSize": undefined,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                      },
                      Object {},
                      Array [
                        Object {
                          "color": "#FFFFFF",
                          "fontSize": 14,
                          "textAlign": "center",
                        },
                        undefined,
                        undefined,
                      ],
                    ]
                  }
                >
                  Add secret account
                </Text>
              </_class>
            </_class>
          </_class>
        </_class>
      </_class>
    </_class>
  </RCTScrollView>
</_class>
`;
